tests/testutil/require.go

```diff
@@ -0,0 +1,253 @@
+		t.Logf("pods-condition=%v\n", podInfo.Status.Conditions)
+	}
+}
+
+func fetchStatefulSet(k8sClient client.Client, name, namespace string) (*appsv1.StatefulSet, error) {
+	ss := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+
+	if err := k8sClient.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: name}, ss); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+
+		return nil, err
+	}
+
+	return ss, nil
+}
+
+func statefulSetIsReady(ss *appsv1.StatefulSet) bool {
+	return ss.Status.ReadyReplicas == ss.Status.Replicas
+}
+
+func RequireDatabaseToFunction(t *testing.T, crdbCluster CockroachCluster) {
+	sqlPort := int32(26257)
+	conn := &database.DBConnection{
+		Ctx:    context.TODO(),
+		Client: crdbCluster.K8sClient,
+		Port:   &sqlPort,
+		UseSSL: true,
+
+		RestConfig:   crdbCluster.Cfg,
+		ServiceName:  fmt.Sprintf("%s-0.%s", crdbCluster.StatefulSetName, crdbCluster.StatefulSetName),
+		Namespace:    crdbCluster.Namespace,
+		DatabaseName: "system",
+
+		RunningInsideK8s:            false,
+		ClientCertificateSecretName: crdbCluster.ClientSecret,
+		RootCertificateSecretName:   crdbCluster.NodeSecret,
+	}
+
+	// Create a new database connection for the update.
+	db, err := database.NewDbConnection(conn)
+	require.NoError(t, err)
+	defer db.Close()
+
+	if _, err := db.Exec("CREATE DATABASE test_db"); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := db.Exec("USE test_db"); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create the "accounts" table.
+	if _, err := db.Exec("CREATE TABLE IF NOT EXISTS accounts (id INT PRIMARY KEY, balance INT)"); err != nil {
+		t.Fatal(err)
+	}
+
+	// Insert two rows into the "accounts" table.
+	if _, err := db.Exec(
+		"INSERT INTO accounts (id, balance) VALUES (1, 1000), (2, 250)"); err != nil {
+		t.Fatal(err)
+	}
+
+	// Print out the balances.
+	rows, err := db.Query("SELECT id, balance FROM accounts")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+	t.Log("Initial balances:")
+	for rows.Next() {
+		var id, balance int
+		if err := rows.Scan(&id, &balance); err != nil {
+			t.Fatal(err)
+		}
+		t.Log("balances", id, balance)
+	}
+
+	countRows, err := db.Query("SELECT COUNT(*) as count FROM accounts")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer countRows.Close()
+	count := getCount(t, countRows)
+	if count != 2 {
+		t.Fatal(fmt.Errorf("found incorrect number of rows.  Expected 2 got %v", count))
+	}
+
+	t.Log("finished testing database")
+}
+
+func getCount(t *testing.T, rows *sql.Rows) (count int) {
+	for rows.Next() {
+		err := rows.Scan(&count)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	return count
+}
+
+func RequireCertificatesToBeValid(t *testing.T, crdbCluster CockroachCluster) {
+	t.Log("Verifying the Certificates")
+	kubectlOptions := k8s.NewKubectlOptions("", "", crdbCluster.Namespace)
+
+	// Get the CA certificate secret and load the ca cert
+	caSecret := k8s.GetSecret(t, kubectlOptions, crdbCluster.CaSecret)
```

Comment:
Maybe put these inside of the `if` below on line 202

Reply:
We need to get the CA cert in both the cases, if ca is user provided or not. As we are using this CA certificate to verify the node certificates are signed by this CA or not at line 217.